libs/community/langchain_community/retrievers/bm25.py

```diff
@@ -97,8 +97,15 @@ def from_documents(
         )
 
     def _get_relevant_documents(
-        self, query: str, *, run_manager: CallbackManagerForRetrieverRun
+        self,
+        query: str,
+        *,
+        run_manager: CallbackManagerForRetrieverRun,
+        k: Optional[int] = None,
```

Comment:
I believe this is supposed to be a non-public method, so I think it would only use self.k